content/en/docs/ops/common-problems/upgrade-issues/index.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Upgrade Problems
+description: Resolve common problems with Istio upgrades.
+weight: 40
+owner: istio/wg-policies-and-telemetry-maintainers
+test: n/a
+---
+
+## `EnvoyFilter` migration
+
+`EnvoyFilter` is an alpha API that is tightly coupled to the implementation
+details of Istio xDS configuration generation. Production use of the alpha API
+must be carefully curated during the upgrade of Istio's control or data plane.
+In many instances, `EnvoyFilter` can be replaced with a first-class Istio API
+which carry substantially lower upgrade risks.
+
+### Use Telemetry API for metrics customization
+
+The usage of `IstioOperator` to customize Prometheus metrics generation has been
+replaced by [Telemetry API](/docs/tasks/observability/metrics/customize-metrics/),
+because `IstioOperator` relies on a template `EnvoyFilter` to change the
+metrics filter configuration.
+
+As an example, the following `IstioOperator` configuration adds a `destination_port` tag:
```

Comment:
Do we need to mention that `MetricDefinition` is not supported by Telemetry API ?

Reply:
I think it's fine to leave that out as the intent is to support it eventually.